modules/augment/include/opencv2/augment/rotate.hpp

```diff
@@ -0,0 +1,55 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+
+#ifndef OPENCV_AUGMENT_ROTATE_HPP
+#define OPENCV_AUGMENT_ROTATE_HPP
+#include <opencv2/augment/Transform.hpp>
+
+namespace cv {
+namespace augment {
+
+    //! @addtogroup augment
+        //! @{
+
+    class CV_EXPORTS_W Rotate : public Transform
+    {
+    public:
+        /* @brief Constructor
+           @param _angleRange the range of angle values to rotate the image
+        */
+        CV_WRAP Rotate(Range _angleRange);
+
+        /* @brief Constructor to initialize the rotation transformation with default angleRange
+        */
+        CV_WRAP Rotate();
+
+        /* @brief Apply the rotation to a single image
+            @param _src Input image to be flipped
+            @param _dst Output (rotated) image
+        */
+        CV_WRAP void image(InputArray _src, OutputArray _dst);
+
+        /* @brief Apply the rotation for a single point
+            @param _src Input point to be rotated
+        */
+        Point2f point(InputArray image,const Point2f& src);
+
+        /* @brief choose an angle from the specified range to apply in next transformations
+        */
+        void resetRandom();
+
+    private :
+        Range angleRange;
+        int currentAngle;
+        RNG rng;
```

Comment:
There is no need to keep it. You can generate random numbers without global random generator.

Reply:
It is actually needed because the random seed is reset with every call to rng.uniform().. otherwise we would need to set it manually each time and we would need another random generator for that :D  

I changed it to be only one static RNG object in the parent Transform class not to waste any memory on keeping one per transformation